test/cql-pytest/test_secondary_index.py

```diff
@@ -241,3 +241,23 @@ def test_filter_cluster_key(cql, test_keyspace):
         stmt = SimpleStatement(f"SELECT c1, c2 FROM {table} WHERE c1 = 1 and c2 = 1 ALLOW FILTERING")
         rows = cql.execute(stmt)
         assert_rows(rows, [1, 1])
+
+# Reproduces #9085.
+def test_multi_column_skips_regular_index(cql, test_keyspace):
```

Comment:
I tested this case against Cassandra (4.0-rc2) and it failed - perhaps you need to add the `scylla_only` fixture?

Reply:
Done.